testsuite/tests/atomic-locs/atomic_fields.ml

```diff
@@ -10,7 +10,7 @@ type t = {
 let get_y (t : t) = t.y
 let set_y (t : t) new_y = t.y <- new_y
 [%%expect{|
-type t = { x : int; mutable y : int; }
+type t = { x : int; mutable(<non-legacy>) y : int [@atomic]; }
```

Comment:
Hmm I thought we agreed that `mutable` will continue to mean `mutable(legacy)`?

Reply:
yes, sorry, this PR is not yet ready for review (that's why it's marked as a draft). I'll make it not a draft once I'm ready for you to review it.